server/src/main/java/com/hedera/block/server/persistence/storage/archive/LocalGroupZipArchiveTask.java

```diff
@@ -0,0 +1,201 @@
+            // handle?
+            FileUtilities.createFile(zipFilePath);
+        }
+        LOGGER.log(Level.DEBUG, "Target Zip Path [%s]".formatted(zipFilePath));
+        // Then, we need to populate the zip with all entries that should be in it, resolved before
+        // the invocation of this method.
+        try (final ZipOutputStream out = new ZipOutputStream(Files.newOutputStream(zipFilePath))) {
+            out.setMethod(ZipOutputStream.STORED);
+            out.setLevel(Deflater.NO_COMPRESSION);
+            for (int i = 0; i < pathsToArchive.size(); i++) {
+                final Path pathToArchive = pathsToArchive.get(i);
+                final String relativizedEntryName =
+                        rootToArchive.relativize(pathToArchive).toString();
+                final ZipEntry zipEntry = new ZipEntry(relativizedEntryName);
+                LOGGER.log(Level.TRACE, "Adding Zip Entry [%s] to zip file [%s]".formatted(zipEntry, zipFilePath));
+                zipEntry.setMethod(ZipEntry.STORED);
+                // @todo(517) we should put a limit to the size that could be read into memory
+                final byte[] blockFileBytes = Files.readAllBytes(pathToArchive);
+                zipEntry.setSize(blockFileBytes.length);
+                zipEntry.setCompressedSize(blockFileBytes.length);
+                final CRC32 crc32 = new CRC32();
+                crc32.update(blockFileBytes);
+                zipEntry.setCrc(crc32.getValue());
+                out.putNextEntry(zipEntry);
+                out.write(blockFileBytes);
+                out.closeEntry();
+                LOGGER.log(
+                        Level.TRACE,
+                        "Zip Entry [%s] successfully added to zip file [%s]".formatted(zipEntry, zipFilePath));
+            }
+        } catch (final IOException e) {
+            // If an exception is thrown here, we need to delete the zip file we just made
+            Files.deleteIfExists(zipFilePath);
+            // We need to propagate the exception in order to not continue with any further executions of this
+            // archiving tasks!
+            throw e;
+        }
+        // If no exception is thrown, this means that the zip is successfully created.
+        LOGGER.log(Level.DEBUG, "Zip File [%s] successfully created".formatted(zipFilePath));
+        return zipFilePath;
+    }
+
+    private static void createLink(final Path rootToArchive, final Path zipFilePath) throws IOException {
+        // We need to create a link to the zip file we just created so readers can find it.
+        final Path livelink = FileUtilities.appendExtension(rootToArchive, ".zip");
+        try {
+            Files.createLink(livelink, zipFilePath);
+        } catch (final Exception e) {
+            LOGGER.log(
+                    Level.ERROR,
+                    "Error while creating link [%s <-> %s], attempting to create a symlink"
```

Comment:
I would avoid the word error here; it causes problems for operations.
```suggestion
                    "Unable to create hard link [%s <-> %s], attempting to create a symbolic link"
```

Reply:
✅ Done, thanks for the suggestion.